programs/server/Server.cpp

```diff
@@ -1119,6 +1119,35 @@ int Server::main(const std::vector<std::string> & /*args*/)
             if (config->has("keeper_server"))
                 global_context->updateKeeperConfiguration(*config);
 
+            /// Reload the number of threads for global pools.
+            /// Note: If you specified it in the top level config (not it config of default profile)
+            /// then ClickHouse will use it exactly.
+            /// This is done for backward compatibility.
+            if (global_context->areBackgroundExecutorsInitialized() && (config->has("background_pool_size") || config->has("background_merges_mutations_concurrency_ratio")))
```

Comment:
not very important, but maybe we can check `global_context->areBackgroundExecutorsInitialized()` only once, and then check all the settings.

Reply:
Btw, if previously the setting was changed in config, and then removed, won't it be more logical to reset default?
After server reload it would use default anyway.